onnxruntime/core/session/inference_session_utils.cc

```diff
@@ -0,0 +1,211 @@
+      session_options.graph_optimization_level = TransformerLevel::Level1;
+      return Status::OK();
+
+    case 2:
+      LOGS(logger, INFO) << "Setting graph_optimization_level to ORT_ENABLE_EXTENDED";
+      session_options.graph_optimization_level = TransformerLevel::Level2;
+      return Status::OK();
+
+    case 3:
+      LOGS(logger, INFO) << "Setting graph_optimization_level to ORT_ENABLE_ALL";
+      session_options.graph_optimization_level = TransformerLevel::Level3;
+      return Status::OK();
+
+    default:
+      LOGS(logger, ERROR) << "Unsupported graph_optimization_level value in ORT config: " << value;
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+                             "Unsupported graph_optimization_level value in ORT config: ", value);
+  }
+}
+
+static Status SetEnableProfiling(SessionOptions& session_options,
+                                 int value,
+                                 const logging::Logger& logger) {
+  if (value != 0 && value != 1) {
+    LOGS(logger, ERROR) << "Unsupported value for enable_profiling option: " << value;
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Unsupported value for enable_profiling option: ", value);
+  }
+
+  LOGS(logger, INFO) << "Setting enable_profiling to " << (value == 0 ? "false" : "true");
+  session_options.enable_profiling = (value == 0 ? false : true);
+  return Status::OK();
+}
+
+//---------------------------------------------------
+//--- end of session options related helpers ---
+//---------------------------------------------------
+
+//---------------------
+//--- end of local helpers ---
+//---------------------
+
+Status InferenceSessionUtils::ParseOrtConfigJsonInModelProto(const ONNX_NAMESPACE::ModelProto& model_proto) {
+  if (is_model_checked_for_ort_config_json_) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "The Model Proto has already been checked for the ORT config json.");
+  }
+
+  for (const auto& metadata_field : model_proto.metadata_props()) {
+    if (metadata_field.has_key() && metadata_field.key() == inference_session_utils::ort_config_key) {
+      LOGS(logger_, INFO)
+          << "Found session/run/environment configuration in the model file to be used while running the model";
+
+      try {
+        parsed_json_ = json::parse(metadata_field.value());
+        // set the flag indicating that the model has the ORT config json.
+        is_json_available_ = true;
+      } catch (const std::exception& e) {
+        LOGS(logger_, ERROR) << "Json stored in the `ort_config` key cannot be parsed. Error message: " << e.what();
+        return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Json stored in the `ort_config` key cannot be parsed. Error message: ", e.what());
+      }
+
+      break;
+    }
+  }
+
+  // all steps complete, set the flag indicating that the model has been checked for the ORT config json.
+  is_model_checked_for_ort_config_json_ = true;
```

Comment:
When you say 'checked', you mean 'read', right?

Reply:
Right. Do you want me to change that to read ?

Reply:
To clarify - The right sequence is: 'check' and then 'read' (if found).

Reply:
Per offline discussion and latest refactor, we are not going to be using a session option, so the naming is moot. I will mark this comment as resolved.